src/audio/asrc/asrc.c

```diff
@@ -354,6 +378,105 @@ static int asrc_params(struct comp_dev *dev,
 	return 0;
 }
 
+static int asrc_dai_find(struct comp_dev *dev,
```

Comment:
This should be a pipeline API call. @bkokoszx I think this also aligns with your verify PR as you have similar logic for finding DAIs ? We should reuse the same function.

Reply:
@bkokoszx Some kind of walk upstream/downstream until component type equals found function would be useful. In this case the walk starts from ASRC. Does your PR #2228 contain such already?

Reply:
@singalsu 
In PR #2228 (feature #1280) I've added function ```pipeline_comp_hw_params()```, which uses already implemented recursive function ```pipeline_for_each_comp()```. It goes from ```HOST``` to ```DAI``` end fetches DAI parameters set during ```dai_set_config()```. I will try to take a look how can we unify both patches.

Reply:
Thanks @bkokoszx !